tenseal/cpp/tensors/ckkstensor.cpp

```diff
@@ -0,0 +1,187 @@
+#include "tenseal/cpp/tensors/ckkstensor.h"
+
+namespace tenseal {
+
+using namespace seal;
+using namespace std;
+
+CKKSTensor::CKKSTensor(const shared_ptr<TenSEALContext>& ctx,
+                       const PlainTensor<double>& tensor,
+                       std::optional<double> scale)
+    : _shape(tensor.shape()), _strides(tensor.strides()) {
+    this->link_tenseal_context(ctx);
+    if (scale.has_value()) {
+        this->_init_scale = scale.value();
+    } else {
+        this->_init_scale = ctx->global_scale();
+    }
+
+    vector<double> plain_data = tensor.data();
+    for (int i = 0; i < plain_data.size(); i++)
+        _data.push_back(
+            CKKSTensor::encrypt(ctx, this->_init_scale, plain_data[i]));
+}
+
+Ciphertext CKKSTensor::encrypt(const shared_ptr<TenSEALContext>& ctx,
```

Comment:
Do we need a dedicated call for vectors of length 1?

The other implementations don't have it, as far as I remember.

Reply:
Not of length 1, but we will need to encrypt vectors when we use batching, and I wanted to use the same function for encrypting a single value, which can be just a vector of length one. Is there a better way to do this?

Reply:
I was just wondering why we need a different call.

We already have encrypt(const shared_ptr<TenSEALContext>& ctx, const double scale, const vector<double>& data)

Reply:
Got you! Yeah you are right!